libs/blocks/region-nav/region-nav.js

```diff
@@ -35,22 +35,28 @@ export function decorateLink(link, path) {
     try { pathname = new URL(pathname).pathname; } catch (e) { /* href does not contain domain */ }
   }
   const linkParts = pathname.split('/');
-  const prefix = linkParts[1] || '';
+  const expressPrefix = linkParts[1] || '';
+  const { languageMap, languages, locales } = getConfig();
+  const language = languages
+    ? getLanguage(languages, locales, pathname) : getLocale(locales, pathname);
+  const prefix = language.prefix.replace('/', '');
+
   let { href } = link;
   if (href.endsWith('/')) href = href.slice(0, -1);
 
-  const { languageMap } = getConfig();
-  if (languageMap && !getConfig().locales[prefix]) {
-    const valueInMap = languageMap[prefix];
+  if (languageMap && !getConfig().locales[expressPrefix]) {
+    const valueInMap = languageMap[expressPrefix];
     href = href.replace(`/${prefix}`, valueInMap ? `/${valueInMap}` : '');
   }
   link.href = `${href}${path}`;
 
+  const interactionPrefix = languageMap ? expressPrefix : prefix;
+
```

Comment:
Hi @yeibercano , it appears interactionPrefix is coming empty and hence for urls that don't exists for that locale, it's pointing to the US home page instead of home page of that locale/language. for example
https://stage--cc--adobecom.hlx.page/creativecloud/animation/discover?milolibs=mwpw-171040#langnav
Hover over Argentina, page doesn't exist for that and it is pointing to US page instead of adobe.com home page for Argentina
<img width="1786" alt="Screenshot 2025-04-27 at 9 02 17 PM" src="https://github.com/user-attachments/assets/7c37bb96-4380-44ac-8c83-0e26cd405a39" />

Proposed solution

```suggestion
 let prefix = '';
  const { languageMap, languages, locales } = getConfig();
  if (languageMap) {
    const linkParts = pathname.split('/');
    prefix = linkParts[1] || '';
  } else {
    const language = languages
      ? getLanguage(languages, locales, pathname) : getLocale(locales, pathname);
    prefix = language.prefix.replace('/', '');
  }
```

This way we won't need to update mouseover event, create new variable. Please review.